### PR TITLE
3.0.0

### PR DESCRIPTION
**Pull request by CFlaniganMide**
This release is a different approach to the python 3 conversion.  It drops support for python 2 entirely, and is overall more stable than the 2.0.0 release.

### CODE REVIEW

File: ebmlite/core.py [hunk 1]
```diff
@@ -1200,16 +1149,16 @@ def addElement(self, eid, ename, baseClass, attribs={}, parent=None,
         """
 
         def _getBool(d, k, default):
-            " Helper function to get a dictionary value cast to bool. "
+            """ Helper function to get a dictionary value cast to bool. """
             try:
-                return str(d[k]).strip()[0] in 'Tt1'
+                return bytearray(d[k]).strip()[0] in b'Tt1'
```
Comment (StokesMIDE) [inline ebmlite/core.py]: Won't this fail if `d[k]` in some cases because there's no encoding specified?

File: ebmlite/core.py [hunk 2]
```diff
@@ -1295,15 +1244,13 @@ def _getInt(d, k, default):
 
         return eclass
 
-
     def __repr__(self):
         try:
-            return "<%s %r from '%s'>" % (self.__class__.__name__, self.name,
-                                          self.filename or self.source)
+            return b"<%s %r from '%s'>" % (self.__class__.__name__, self.name,
```
Comment (StokesMIDE) [inline ebmlite/core.py:1249]: I think `__repr__()` is supposed to return a string.

File: ebmlite/core.py (hunk 1 above)
Comment (CFlaniganMide) [reply ebmlite/core.py]: I actually just ran into this, and yes is does.  I put a fix in, but I'm not 100% happy with it.

File: ebmlite/core.py (hunk 2 above)
Comment (CFlaniganMide) [reply ebmlite/core.py:1249]: good catch, I'll check all the __repr__/__str__ methods

**Pull request closed without merging**